nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -112,19 +117,60 @@ public boolean canPurgeOrUntrash(List<DocumentModel> docs, NuxeoPrincipal princi
             return false;
         }
         // used to do only check on parent perm
-        TrashInfo info = getInfo(docs, principal, false, true);
+        TrashInfo info = getInfo(docs, principal, false, true, true);
         return info.docs.size() == docs.size();
     }
 
     protected TrashInfo getInfo(List<DocumentModel> docs, NuxeoPrincipal principal, boolean checkProxies,
             boolean checkDeleted) {
+        return getInfo(docs, principal, checkProxies, checkDeleted, false);
+    }
+
+    @Override
+    public boolean getCanEmptyTrash(DocumentModel parent, NuxeoPrincipal principal) {
+        String nxql = String.format(TRASHED_QUERY, parent.getId());
+        ScrollRequest request = DocumentScrollRequest.builder(nxql)
+                                                     .username(principal.getName())
+                                                     .repository(parent.getRepositoryName())
+                                                     .build();
+        ScrollService service = Framework.getService(ScrollService.class);
+        boolean canPurge = true;
+        try (Scroll scroll = service.scroll(request)) {
+            if (!scroll.hasNext()) {
+                return false;
+            }
+            while (scroll.hasNext() && canPurge) {
+                DocumentModelList documents = RepositoryScroll.loadDocuments(parent.getCoreSession(), scroll.next());
+                canPurge = canPurgeOrUntrash(documents, principal);
+            }
+        }
+        return canPurge;
+    }
+
+    /**
+     * Return the {@link TrashInfo} of a document list with potentially some documents not ok for 'purge' or unTrash
+     * (forbidden, locked ...)
+     *
+     * @param docs the documents List to trash
+     * @param principal the user principal
+     * @param checkProxies <code>true</code> to check proxies.
+     * @param checkDeleted <code>true</code> to check if doc is trashed.
+     * @param exitOnFirstDelta <code>true</code> to avoid looping in the whole documents List and exist on first
+     *            document with TrashInfo "forbidden", "proxies" or "locked".
+     * @return the TrashInfo for the documents List.
+     */
+    protected TrashInfo getInfo(List<DocumentModel> docs, NuxeoPrincipal principal, boolean checkProxies,
+            boolean checkDeleted, boolean exitOnFirstDelta) {
```

Comment:
we dont change the protected/public methods signature for backward compatibility just create a new once and call it from the exiting once